src/ILVerify/src/ILImporter.Verify.cs

```diff
@@ -1208,7 +1218,12 @@ void ImportStoreIndirect(TypeDesc type)
 
         void ImportThrow()
         {
-            throw new NotImplementedException();
+            var value = Pop();
+
+            if (value.Kind != StackValueKind.ObjRef)
+            {
+                VerificationError(VerifierError.StackNotEq);
```

Comment:
`Expected an ObjRef on the stack` would be more appropriate message for this case.